cpp/models/ide_secir/parameters.h

```diff
@@ -36,87 +38,453 @@ namespace isecir
      */
-    DelayDistribution(ScalarType init_max_support)
-        : m_max_support{init_max_support}
+    ExponentialDecay(ScalarType init_funcparam)
+        : StateAgeFunction(init_funcparam)
     {
     }
+
     /**
-     * @brief DelayDistribution is currently defined as a smoothed cosine function.
-     * 
-     * Used function goes through points (0,1) and (m_max_suppor,0) and is interpolated in between using a smoothed cosine function.
+     * @brief Defines exponential decay function depending on state_age.
+     *
+     * The parameter m_funcparam defines how fast the exponential function decays.
      * 
      * @param state_age time at which the function should be evaluated
-     * @return ScalarType evaluation of the smoother cosine function
+     * @return ScalarType evaluation of the function at state_age. 
      */
-    ScalarType Distribution(ScalarType state_age)
+    ScalarType eval(ScalarType state_age) override
     {
-        return smoother_cosine(state_age, 0.0, m_max_support, 1.0, 0.0);
+        return std::exp(-m_funcparam * state_age);
     }
 
+protected:
     /**
-     * @brief Get the m_max_support object
+     * @brief Implements clone for ExponentialDecay.
      * 
-     * Can be used to access the m_max_support object, which specifies the right bound of the support of the function.
+     * @return StateAgeFunction* 
+     */
+    StateAgeFunction* clone_impl() const override
+    {
+        return new ExponentialDecay(*this);
+    }
+};
+
+/**
+ * @brief Class that defines an smoother cosine function depending on the state age.
+ */
+struct SmootherCosine : public StateAgeFunction {
+
+    /**
+    * @brief Default constructor of the class.
+    */
+    SmootherCosine()
+        : StateAgeFunction()
+    {
+    }
+
+    /**
+     * @brief Constructs a new SmootherCosine object
      * 
-     * @return ScalarType 
+     * @param[in] init_funcparam specifies the initial function parameter of the function.
      */
-    ScalarType get_max_support() const
+    SmootherCosine(ScalarType init_funcparam)
+        : StateAgeFunction(init_funcparam)
     {
-        return m_max_support;
     }
 
     /**
-     * @brief Set the m_max_support object
+     * @brief Defines smoother cosine function depending on state_age.
+     *
+     * Used function goes through points (0,1) and (m_funcparam,0) and is interpolated in between using a smoothed cosine function.
      * 
-     * Can be used to set the m_max_support object, which specifies the right bound of the support of the function.
+     * @param[in] state_age time at which the function should be evaluated
+     * @return ScalarType evaluation of the function at state_age. 
+     */
+    ScalarType eval(ScalarType state_age) override
+    {
+        return smoother_cosine(state_age, 0.0, m_funcparam, 1.0, 0.0);
+    }
+
+    ScalarType get_max_support(ScalarType dt, ScalarType tol = 1e-10) override
+    {
+        unused(dt);
+        unused(tol);
```

Comment:
yes

Reply:
I actually undid this change again because this led to problems when getting the maximum support of a StateAgeFunction directly, e.g.
`SmootherCosine smoothcos(2.0);
smoothcos.get_maximum_support()`
because two arguments were expected. For some reason it didn't lead to errors when calling doing this via SAFWrapper.